src/pybind/mgr/dashboard/controllers/rbd.py

```diff
@@ -342,13 +342,30 @@ def _copy(d_ioctx):
                 # Set features
                 feature_bitmask = _format_features(features)
 
+                if snapshot_name:
+                    s_img.set_snap(snapshot_name)
+
                 s_img.copy(d_ioctx, dest_image_name, feature_bitmask, l_order,
                            stripe_unit, stripe_count, data_pool)
 
             return _rbd_call(dest_pool_name, _copy)
 
         return _rbd_image_call(pool_name, image_name, _src_copy)
 
+    @RbdTask('flatten', ['{pool_name}', '{image_name}'], 2.0)
+    @RESTController.resource(['POST'])
+    def flatten(self, pool_name, image_name):
+
+        def _flatten(ioctx, image):
+            image.flatten()
+
+        return _rbd_image_call(pool_name, image_name, _flatten)
+
+    @RESTController.collection()
```

Comment:
please specify the HTTP method allowed (I guess it should be GET): `@RESTController.collection(['GET'])`